sdk/storage/azure-storage-common/inc/azure/storage/common/paged_response.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cstdlib>
+#include <string>
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/raw_response.hpp>
+
+namespace Azure {
+
+template <class Derived> class PagedResponse {
+public:
+  std::string NextPageToken;
+  std::string CurrentPageToken;
+  std::unique_ptr<Azure::Core::Http::RawResponse> RawResponse;
+
+  bool HasMore() const { return m_hasMore; }
+
+  void NextPage(const Azure::Core::Context& context)
```

Comment:
nit: I have been thinking about our API names here, emphasizing pattern discovery and self-documenting, especially since the type isn't yet iteratable.

I would prefer we rename these to `HasMorePages()` and `GetNextPage()`. @JeffreyRichter would you be OK with that?

Reply:
OK, the `Get` prefix isn't great here since `NextPage()` returns void. How about something like `MoveNext()` or `MovetoNextPage()`?

Reply:
Yes, I'm OK with these name changes.

Reply:
@ahsonkhan Which one of these two functions are customers supposed to call first? Checking if `HasMorePages()` before `MoveToNextPage()` or the other way around?

Reply:
`HasMorePages()` first, then `MoveToNextPage()`

Basically:
```C++
if (HasMorePages())
{
  MovetoNextPage(); // This sets HasMorePages to false if there are no more pages left
}
```

Which one of these two would you prefer as the name, Jinming: `MoveNext()` OR `MoveToNextPage()`?

I think both are good (so I leave it up to you to decide), but I lean _slightly_ towards `MoveNext()`, if you are on the fence (but I am a bit biased since that's [the name used in .NET enumerator](https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerator.movenext?view=net-5.0)).

Reply:
I also prefer `HasMorePages` being called before `MoveToNextPage`. But this is kind of misleading to customers.

For example, the user sends a list request and server returns all the results without continuation token.
Now the user gets a `PagedResponse` with all the results in it.
After processing this page, he will call `HasMorePages`, this function will return `true`. So the user will call `MoveToNextPage()`.
But `MoveToNextPage()` won't send another request because the continuation token is empty. [code](https://github.com/Azure/azure-sdk-for-cpp/blob/dab510067f5bfc41c11dd054a1ab4414e6ffe42f/sdk/storage/azure-storage-common/inc/azure/storage/common/paged_response.hpp#L28)
Now the user will wrongly think he has got a new page, but actually he didn't. It's still the first page.

Reply:
> Which one of these two would you prefer as the name, Jinming: MoveNext() OR MoveToNextPage()?

I prefer `MoveToNextPage()`. because `MoveToNextPage` and `HasMorePages` look like a pair. `HasMore` and `MoveNext` are a pair.